### DISCUSSION TITLE
Database connection for models of different packages

### DISCUSSION THREAD

**Discussion by nick-nds** [spatie/laravel-multitenancy#469] [Q&A]
Hello,
I noticed in the documentation that every model should use UsesLandlordConnection or UsesTenantConnection. So now I am using a package and I am unsure how to add these traits to their models. I have tried changing the default connection to tenant or landlord using middleware but I am not sure if this is the right way.

**Comment by citricguy** [comment]
I'm running into the same issue, did you by chance resolve this?

**Reply by nick-nds** [reply]
No, not exactly. The package had the option to set Model connection through config values. I changed the database connection config for that package to use the tenant connection.

**Comment by kieranbarlow** [comment]
Within the model you should import the class from the package.

`use Spatie\Multitenancy\Models\Concerns\UsesTenantConnection;`

Once imported, specify that your model uses it.

`use UsesTenantConnection;`

**Reply by nick-nds** [reply]
How can we add this trait to models of other packages?

**Reply by kieranbarlow** [reply]
You would extend the models of other packages. 

For example, if you are using Spatie/Laravel-Permission package. You would want to adjust the Role & Permission models to also look at UsesTenantConnection.

Example Role model extended from the original package model.

```
<?php

namespace App\Models;

use Spatie\Multitenancy\Models\Concerns\UsesTenantConnection;
use Spatie\Permission\Models\Role as SpatieRole;
use Spatie\Permission\Contracts\Role as RoleContract;

class Role extends SpatieRole implements RoleContract
{
    use UsesTenantConnection;
}
```

**Reply by nick-nds** [reply]
ok.. So if this Role model is being used internally by spatie/laravel-permission, they will still be using the default Role model instead of our extended \App\Models\Role. Therefore spatie is still using the Role model with default connection instead of using Tenant Connection.

**Reply by kieranbarlow** [reply]
It will depend on the package you are using but Spatie/Laravel-Permission allows you to determine the "Model" used. Whether that is your extended model or the default package model...

Below found within permission.php config file. 

```
<?php

return [

    'models' => [

        /*
         * When using the "HasPermissions" trait from this package, we need to know which
         * Eloquent model should be used to retrieve your permissions. Of course, it
         * is often just the "Permission" model but you may use whatever you like.
         *
         * The model you want to use as a Permission model needs to implement the
         * `Spatie\Permission\Contracts\Permission` contract.
         */

        'permission' => App\Models\Permission::class,

        etc......
```

**Reply by nick-nds** [reply]
Yes true. But not all packages give this option. So we are back to square one.

**Reply by kieranbarlow** [reply]
In that case you would fork the package and adjust as you need.

**Reply by nick-nds** [reply]
Exactly. That's what I said/meant.
